### PR TITLE
chore(STONEO11Y-52): Setup pint for the CI

### PR DESCRIPTION
**Pull request by eisraeli**
Pint[1] is a Prometheus rule linter.

[1] https://github.com/cloudflare/pint

### CODE REVIEW

File: .tekton/pull-request.yaml [hunk 1]
```diff
@@ -37,12 +37,25 @@ spec:
...
             workspace: workspace
         runAfter:
           - clone-repository
+        taskSpec:
+          workspaces:
+            - name: workspace
+          steps:
+            - name: lint-rules
+              image: registry.access.redhat.com/ubi8/ubi:latest
+              script: cd $(workspaces.workspace.path) && ./automation/lint_prometheus-rules.sh
```
Comment (amisstea) [inline .tekton/pull-request.yaml:52]: This works but you can also change the starting directory using the `workingDir` field for a step.

**Review by amisstea — APPROVED** [review]

**Comment by amisstea** [comment]
@eisraeli do you plan to fix the linting issues `pint` found?

File: .tekton/pull-request.yaml (hunk 1 above)
Comment (eisraeli) [reply .tekton/pull-request.yaml]: Indeed. I've changed it to work with `workingDir`

**Comment by eisraeli** [comment]
> @eisraeli do you plan to fix the linting issues `pint` found?

Not yet. I need to look into it.

File: .tekton/pull-request.yaml [hunk 2]
```diff
@@ -37,12 +37,26 @@ spec:
...
+            - name: lint-rules
+              image: registry.access.redhat.com/ubi8/ubi:latest
+              workingDir: $(workspaces.workspace.path)
+              script: ./automation/lint_prometheus-rules.sh
+      - name: run-tests
+        workspaces:
+          - name: workspace
+            workspace: workspace
+        runAfter:
+          - lint-prometheus-rules
```
[the remark this answers is not in the document]
Comment (eisraeli) [reply .tekton/pull-request.yaml:59]: @yftacherzog Yeah, you are right. The can run in parallel after `clone-repository`. Changed.

**Comment by eisraeli** [comment]
> Structure looks good to me. Just need to make the test pass and make the call in such a way that allows additional rule files to be added.

@yftacherzog 
1. The lint pass successfully if only provide with the `groups` stanza, similar to the provided examples.

2. That can be easily achieved once we have multiple rules files.
If we expect multiple rules files, we should create a Prometheus rules folder.

**Comment by yftacherzog** [comment]
> > Structure looks good to me. Just need to make the test pass and make the call in such a way that allows additional rule files to be added.
> 
> @yftacherzog
> 
> 1. The lint pass successfully if only provide with the `groups` stanza, similar to the provided examples.
> 2. That can be easily achieved once we have multiple rules files.
>    If we expect multiple rules files, we should create a Prometheus rules folder.

You're right. No need for multiple files for now.

I didn't understand the part with the missing stanza. What do we need to do so that this is passing and the PR is ready to be merged?

**Comment by eisraeli** [comment]
> > > Structure looks good to me. Just need to make the test pass and make the call in such a way that allows additional rule files to be added.
> > 
> > 
> > @yftacherzog
> > 
> > 1. The lint pass successfully if only provide with the `groups` stanza, similar to the provided examples.
> > 2. That can be easily achieved once we have multiple rules files.
> >    If we expect multiple rules files, we should create a Prometheus rules folder.
> 
> You're right. No need for multiple files for now.
> 
> I didn't understand the part with the missing stanza. What do we need to do so that this is passing and the PR is ready to be merged?

It's not a missing stanza.
The example provided by Cloudfare only consists of the `groups` stanza.
Meaning, that if you remove lines 1-8, you will be left only with the `groups` stanza and the linting will pass successfully.

**Comment by eisraeli** [comment]
> > > > > Structure looks good to me. Just need to make the test pass and make the call in such a way that allows additional rule files to be added.
> > > > 
> > > > 
> > > > @yftacherzog
> > > > 
> > > > 1. The lint pass successfully if only provide with the `groups` stanza, similar to the provided examples.
> > > > 2. That can be easily achieved once we have multiple rules files.
> > > >    If we expect multiple rules files, we should create a Prometheus rules folder.
> > > 
> > > 
> > > You're right. No need for multiple files for now.
> > > I didn't understand the part with the missing stanza. What do we need to do so that this is passing and the PR is ready to be merged?
> > 
> > 
> > It's not a missing stanza. The example provided by Cloudfare only consists of the `groups` stanza. Meaning, that if you remove lines 1-8, you will be left only with the `groups` stanza and the linting will pass successfully.
> 
> ok. Now I understand what you mean. The tool handles prometheus configs rather than k8s configs. It's the exact same situation that we have with promtool now that we're trying to run the unit tests using our own rules.
> 
> Please have a look here: https://github.com/redhat-appstudio/o11y/pull/9/checks I think this should address your problem as well.

@yftacherzog You wrote this here and I wrote the same thing on the promtool PR   :)

**Comment by yftacherzog** [comment]
> @yftacherzog You wrote this here and I wrote the same thing on the promtool PR :)

can you evaluate whether we should use pint or promtool for the purpose of linting?

**Comment by Omeramsc** [comment]
> > Structure looks good to me. Just need to make the test pass and make the call in such a way that allows additional rule files to be added.
> 
> @yftacherzog
> 
> 1. The lint pass successfully if only provide with the `groups` stanza, similar to the provided examples.
> 2. That can be easily achieved once we have multiple rules files.
>    If we expect multiple rules files, we should create a Prometheus rules folder.

We already *kind-of* have one https://github.com/redhat-appstudio/o11y/tree/main/prometheus/base

**Comment by eisraeli** [comment]
> > @yftacherzog You wrote this here and I wrote the same thing on the promtool PR :)
> 
> can you evaluate whether we should use pint or promtool for the purpose of linting?

@yftacherzog I didn't find any articles comparing them and the linting I've run was pretty basic. 
I suggest merging them both and comparing the linting results.
Then we can make a wiser decision if to keep both or just one of them.

**Comment by yftacherzog** [comment]
> > > @yftacherzog You wrote this here and I wrote the same thing on the promtool PR :)
> > 
> > 
> > can you evaluate whether we should use pint or promtool for the purpose of linting?
> 
> @yftacherzog I didn't find any articles comparing them and the linting I've run was pretty basic. I suggest merging them both and comparing the linting results. Then we can make a wiser decision if to keep both or just one of them.

sounds good!

**Review by Omeramsc — APPROVED** [review]
LGTM

**Pull request merged**